File: exercism/ecmascript/bob/bob.js
class Bob {
  hey(msg) {
    if(msg === msg.toUpperCase() && msg.match(/[A-Z]/) !== null) return 'Whoa, chill out!'
    if(msg.match(/[?]$/)) return 'Sure.'
    if(msg.match(/[^ ]/)) return 'Whatever.' 
    return 'Fine. Be that way!'
  }
}

export default Bob;

File: exercism/ruby/bob/bob.rb
class Bob
  def self.hey(remark)
    if remark.match(/^ $/)
      "Whatever."
    elsif remark == remark.upcase && remark[-1] == "?"
      "Calm down, I know what I'm doing!"
    elsif remark[-1] == "?"
      "Sure."
    elsif remark == remark.upcase
      "Whoa, chill out!"
    elsif remark.match(/[^ ]/)
      "Whatever."
    else
      "Fine. Be that way!"
    end
  end
end
